storm-core/test/jvm/org/apache/storm/scheduler/resource/TestResourceAwareScheduler.java

```diff
@@ -54,6 +61,681 @@
+            Assert.assertTrue(entry.getKey()- entry.getValue() >= 0);
+        }
+        Assert.assertEquals("Running - Fully Scheduled by DefaultResourceAwareStrategy", cluster.getStatusMap().get(topology1.getId()));
+    }
+
+    @Test
+    public void testScheduleResilience() {
+        INimbus iNimbus = new TestUtilsForResourceAwareScheduler.INimbusTest();
+        Map<String, Number> resourceMap = new HashMap<>();
+        resourceMap.put(Config.SUPERVISOR_CPU_CAPACITY, 400.0);
+        resourceMap.put(Config.SUPERVISOR_MEMORY_CAPACITY_MB, 2000.0);
+        Map<String, SupervisorDetails> supMap = TestUtilsForResourceAwareScheduler.genSupervisors(2, 2, resourceMap);
+
+        TopologyBuilder builder1 = new TopologyBuilder();
+        builder1.setSpout("wordSpout1", new TestWordSpout(), 3);
+        StormTopology stormTopology1 = builder1.createTopology();
+        Config config1 = new Config();
+        config1.putAll(defaultTopologyConf);
+        Map<ExecutorDetails, String> executorMap1 = TestUtilsForResourceAwareScheduler.genExecsAndComps(stormTopology1, 3, 0);
+        TopologyDetails topology1 = new TopologyDetails("topology1", config1, stormTopology1, 3, executorMap1, 0);
+
+        TopologyBuilder builder2 = new TopologyBuilder();
+        builder2.setSpout("wordSpout2", new TestWordSpout(), 2);
+        StormTopology stormTopology2 = builder2.createTopology();
+        Config config2 = new Config();
+        config2.putAll(defaultTopologyConf);
+        // memory requirement is large enough so that two executors can not be fully assigned to one node
+        config2.put(Config.TOPOLOGY_COMPONENT_RESOURCES_ONHEAP_MEMORY_MB, 1280.0);
+        Map<ExecutorDetails, String> executorMap2 = TestUtilsForResourceAwareScheduler.genExecsAndComps(stormTopology1, 2, 0);
+        TopologyDetails topology2 = new TopologyDetails("topology2", config2, stormTopology2, 2, executorMap2, 0);
+
+        // Test1: When a worker fails, RAS does not alter existing assignments on healthy workers
+        Cluster cluster = new Cluster(iNimbus, supMap, new HashMap<String, SchedulerAssignmentImpl>(), config1);
+        ResourceAwareScheduler rs = new ResourceAwareScheduler();
+        Map<String, TopologyDetails> topoMap = new HashMap<>();
+        topoMap.put(topology2.getId(), topology2);
+        Topologies topologies = new Topologies(topoMap);
+
+        rs.prepare(config1);
+        rs.schedule(topologies, cluster);
+
+        SchedulerAssignmentImpl assignment = (SchedulerAssignmentImpl)cluster.getAssignmentById(topology2.getId());
+        // pick a worker to mock as failed
+        WorkerSlot failedWorker = new ArrayList<WorkerSlot>(assignment.getSlots()).get(0);
+        Map<ExecutorDetails, WorkerSlot> executorToSlot = assignment.getExecutorToSlot();
+        List<ExecutorDetails> failedExecutors = new ArrayList<>();
+        for (Map.Entry<ExecutorDetails, WorkerSlot> entry : executorToSlot.entrySet()) {
```

Comment:
There is a method in SchedulerAssignmentImpl call getSlotToExecutors that you can use